src/lkl/virtio_netdev.c

```diff
@@ -12,27 +12,19 @@
 #define MAX_NET_DEVS 16
 
 static uint8_t registered_dev_idx = 0;
-static uint8_t _netdev_base_id = 0;
 
 struct virtio_dev* registered_devs[MAX_NET_DEVS];
 static struct ticketlock __event_notifier_lock;
 
-/*
- * Function to get the netdev id maintained in virtio_netdev list
- */
-static inline uint8_t get_netdev_id(uint8_t dev_id)
-{
-    return (dev_id - _netdev_base_id);
-}
-
 /*
  * Function to get netdev instance to use its attributes
  */
 static inline struct virtio_dev* get_netdev_instance(uint8_t netdev_id)
 {
-    struct virtio_dev* ndev = registered_devs[netdev_id];
-    assert(ndev != NULL);
-    return ndev;
+    for (size_t i = 0; i < registered_dev_idx; i++)
```

Comment:
@prp this is part of the config PR. I removed the duplication of the logic that determines network device IDs because they could get out of sync, when the host mounts more disks than the enclave expects (e.g. containers/cc test). It now uses the host's IDs, which is more robust and I think there are no security concerns with that.

Reply:
The disks inside/outside should match. The test is incorrect and should be fixed.

Reply:
We can't assume that they will always match, but yes, the test should be fixed too.

Reply:
Why not? If they don't just fail.

Reply:
Yeah, we could just let it fail. It's nicer to just use the host's IDs I think, because it doesn't require us to duplicate the ID-assignment logic.